app/scripts/modules/amazon/src/function/function.transformer.ts

```diff
@@ -1,7 +1,81 @@
-import { IAmazonFunction } from 'amazon/domain';
+import { AWSProviderSettings } from 'amazon/aws.settings';
+import { Application } from '@spinnaker/core';
+
+import { IAmazonFunctionUpsertCommand, IAmazonFunction } from 'amazon/domain';
 
 export class AwsFunctionTransformer {
   public normalizeFunction(functionDef: IAmazonFunction): IAmazonFunction {
-    return functionDef;
+    const normalizedFunctionDef: IAmazonFunction = functionDef;
+    normalizedFunctionDef.credentials = functionDef.account;
+    return normalizedFunctionDef;
+  }
+
+  public convertFunctionForEditing(functionDef: IAmazonFunction): IAmazonFunctionUpsertCommand {
```

Comment:
nit: Using object structuring here can get the job done in a more readable way. Then you can override/delete the attributes that are not consistent with the return you want. 

```
public convertFunctionForEditing = (functionDef: IAmazonFunction): IAmazonFunctionUpsertCommand => ({
  ...functionDef,
   envVariables: functionDef.environment ? functionDef.environment.variables : {},
   others,
})
```

Reply:
done. will appear in next push.